src/options/options.js

```
@@ -386,22 +568,37 @@ function populatePromptTabs() {
-            checkForChanges();
+            
+            // Only trigger changes if we actually switched to a different prompt
+            if (previousActivePrompt !== activePrompt) {
+                updatePromptTabUI();
+                // Update text area and language select with new prompt data
+                promptTextArea.value = systemPrompts[activePrompt].text || "";
+                languageSelect.value = systemPrompts[activePrompt].language || "English";
+                
+                // Show/hide restore button based on whether it's a default prompt
+                restoreDefaultPromptButton.style.display = 
+                    defaultPrompts.hasOwnProperty(activePrompt) ? "inline-block" : "none";
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)